controller/common/src/Controller/Common/Order/Standard.php

```diff
@@ -347,7 +348,13 @@ protected function updateStock( \Aimeos\MShop\Order\Item\Iface $orderItem, int $
 
 				foreach( $items as $item )
 				{
-					$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					if( $item->getType() === 'select' ) {
+						$productItem = $productManager->get( $item->getProductId(), array( 'product' ) );
+						$productRefItem = $productItem->getRefItems( 'product', 'default', 'default' )->where('product.code','=',$item->getProductCode())->first();
+						$stockManager->decrease( [$productRefItem->getId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					} else {
+						$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					}
```

Comment:
Try this:
```php
$prodId = $item->getProductId();

if( $item->getType() === 'select' )
{
	$prodId = $productManager->get( $item->getProductId(), ['product' => ['default']] )
		->getRefItems( 'product', null, 'default' )
		->where( 'product.code', '==', $item->getProductCode() )
		->getId()->first();
}

$stockManager->decrease( [$prodId => $how * -1 * $item->getQuantity()], $item->getStockType() );
```
This fixes a few potential problems in your code.

Reply:
updated the code, tested manually, it works. Thanks for the guide.

how do i adapt the test? 

![image](https://user-images.githubusercontent.com/82187386/125225413-d8667d80-e301-11eb-8a10-3bcd3a81ac7b.png)